src/Catel.MVVM/Catel.MVVM.Shared/MVVM/ViewModels/ViewModelBase.cs

```diff
@@ -1411,10 +1417,19 @@ protected virtual Task InitializeAsync()
 
             lock (_modelLock)
             {
+                bool isDirty = false;
+
                 foreach (var modelKeyValuePair in _modelObjects)
                 {
                     UninitializeModelInternal(modelKeyValuePair.Key, modelKeyValuePair.Value, ModelCleanUpMode.CancelEdit);
+                    var model = modelKeyValuePair.Value as IModel;
```

Comment:
My initial thought is to only change IsDirty on view model property changes. This makes it a bit hard when ending edit on models with mapped properties, but if it's *really* important for the end-user to have the right state, they can always implement something themselves.